exec/java-exec/src/main/java/org/apache/drill/exec/planner/sql/SchemaUtilites.java

```diff
@@ -149,6 +168,16 @@ public static void throwSchemaNotFoundException(final SchemaPlus defaultSchema,
         .build(logger);
   }
 
+  /** Utility method to throw {@link UserException} with context information */
+  public static void throwSchemaNotFoundException(final String defaultSchema, final String givenSchemaPath) {
+    throw UserException.validationError()
+        .message("Schema [%s] is not valid with respect to either root schema or current default schema.",
```

Comment:
Because names can contain dots, we must more careful format the schema. The schema should consists of a list (array) of name parts. For each name:

* If the name contains a dot, wrap it in back ticks: `` `a.b` ``
* Otherwise, use the name as is: `c`
* Concatenate the name parts with dots: `` `a.b`.c ``

Code to do this might already exist. @vvysotskyi may know.